File: android-katha-ai/app/src/main/java/com/rork/kathaai/model/Prompt12Models.kt
package com.rork.kathaai.model

import kotlinx.serialization.Serializable

@Serializable
data class StreakDay(
    val date: Long,
    val active: Boolean,
    val freezeUsed: Boolean = false,
    val summary: String = ""
)

@Serializable
data class StreakState(
    val current: Int = 0,
    val longest: Int = 0,
    val lastActivityDate: Long? = null,
    val nextCreditIn: Int = 3,
    val freezesAvailable: Int = 0,
    val history: List<StreakDay> = emptyList()
)

@Serializable
data class NotificationPreferences(
    val storyNewChapters: Boolean = true,
    val storyNewStories: Boolean = true,
    val storyComments: Boolean = true,
    val storyLikes: Boolean = false,
    val streakReminders: Boolean = true,
    val milestoneCelebrations: Boolean = true,
    val creditsEarned: Boolean = true,
    val weeklyDigest: Boolean = true,
    val newWriters: Boolean = false
)

@Serializable
data class ReferralRecord(
    val id: String,
    val displayName: String,
    val username: String,
    val joinedAt: Long,
    val firstStoryGeneratedAt: Long? = null,
    val credited: Boolean = false
)

@Serializable
data class OfflineStoryRecord(
    val storyId: String,
    val title: String,
    val sizeMb: Double,
    val downloadedAt: Long
)

sealed class SleepTimerOption {
    data object OFF : SleepTimerOption()
    data object END_OF_CHAPTER : SleepTimerOption()
    data class MINUTES(val minutes: Int) : SleepTimerOption()
}

File: ios-katha-ai-create-stories/KathaAICreateStories/Models/Prompt12Models.swift
import Foundation

struct StreakDay: Codable, Hashable, Identifiable {
    let date: Date
    let active: Bool
    let freezeUsed: Bool
    let summary: String

    var id: Date { date }
}

struct StreakState: Codable, Hashable {
    var current: Int = 0
    var longest: Int = 0
    var lastActivityDate: Date?
    var nextCreditIn: Int = 3
    var freezesAvailable: Int = 0
    var history: [StreakDay] = []

    static let initial = StreakState()
}

struct NotificationPreferences: Codable, Hashable {
    var storyNewChapters = true
    var storyNewStories = true
    var storyComments = true
    var storyLikes = false
    var streakReminders = true
    var milestoneCelebrations = true
    var creditsEarned = true
    var weeklyDigest = true
    var newWriters = false
}

struct ReferralRecord: Codable, Hashable, Identifiable {
    let id: String
    let displayName: String
    let username: String
    let joinedAt: Date
    var firstStoryGeneratedAt: Date?
    var credited: Bool
}

struct OfflineStoryRecord: Codable, Hashable, Identifiable {
    let storyId: String
    let title: String
    let sizeMB: Double
    let downloadedAt: Date

    var id: String { storyId }
}

enum SleepTimerOption: Equatable {
    case off
    case endOfChapter
    case minutes(Int)

    var label: String {
        switch self {
        case .off: return "Off"
        case .endOfChapter: return "End of chapter"
        case .minutes(let minutes): return "\(minutes)m"
        }
    }
}
